draft-bbs-signatures.md

```diff
@@ -498,22 +496,28 @@ Procedure:
 
 ### ProofGen
 
-This operation computes a zero-knowledge proof-of-knowledge of a signature, while optionally selectively disclosing from the original set of signed messages. The "prover" may also supply a presentation header, see [presentation header selection](#presentation-header-selection) for more details.
-
-If an application chooses to pass the indexes of the generators instead, then it will also need to pass the indexes of the generators corresponding to the revealed messages.
+This operation computes a zero-knowledge proof-of-knowledge of a signature, while optionally selectively disclosing from the original set of signed messages. The "prover" may also supply a presentation header, see [presentation header selection](#presentation-header-selection) for more details. The generators and messages are supplied as pairs in two different lists: the first list containing (generator, message) pairs with signed messages that will be revealed, and the second list containing (generator, message) pairs corresponding to the signed messages that will be kept hidden. The union of those two lists MUST equal the list of (generator, message) pairs supplied to Sign (i.e., the second list MUST contain all (generator, message) pairs corresponding to signed messages that do not appear in the first list).
 
 ```
-proof = ProofGen(PK, signature, header, ph, (msg_1,..., msg_L), (H_1,..., H_L), RevealedIndexes)
+proof = ProofGen(PK, signature,
+                 [(H_i1, msg_i1), ..., (H_iR, msg_iR)], 
+                 [(H_h1, msg_h1), ..., (H_hU, msg_hU)],
+                 header, ph)
 
 Inputs:
 
 - PK (REQUIRED), an octet string of the form outputted by the SkToPk operation.
 - signature (REQUIRED), an octet string of the form outputted by the Sign operation.
+- [(H_i1, msg_i1), ..., (H_iR, msg_iR)] (REQUIRED), Generators and message pairs 
+                                        corresponding to messages to be revealed. 
+                                        Generators must be G1 points, and the 
+                                        messages must be non-zero scalars mod q.
+- [(H_h1, msg_h1), ..., (H_hU, msg_hU)] (REQUIRED), Generators and messages pairs 
```

Comment:
Not sure why these two are required and can't default to [] as well

Reply:
My thinking was that if both lists are optional, it would allow both to be empty at the same time (which it should not if the original signed list is also non-empty). 

That said, I added a not in the description of ProofGen to highlight this fact and made the lists optional. I agree that this way is better from having the lists be required.